### PR TITLE
Fix the shakedown tests for strict mode

### PR DESCRIPTION
**Pull request by susanxhuynh**
- Temporarily using a stub-universe for HDFS until strict mode support is released, just to get Framework Test to pass. Should be reverted when a new version of HDFS is released.
- Specify the principal and secret when install HDFS in strict mode.
- Specify security configs for Spark driver in strict mode.

### CODE REVIEW

**Comment by susanxhuynh** [comment]
I removed the hdfs master stub repo. The hdfs dependence is currently in the test setup_module() method. The test currently fails during the HDFS install, before any tests are run. Are you suggesting moving that into the Teragen test, and disabling that test temporarily?  I could do that, but since the new hdfs will be released anytime now, I was thinking just wait for it to be released and not disable any tests.

File: tests/test.py [hunk 1]
```diff
@@ -56,7 +57,10 @@ def _require_package(pkg_name):
     pkg_manager = dcos.package.get_package_manager()
     installed_pkgs = dcos.package.installed_packages(pkg_manager, None, None, False)
     if not any(pkg['name'] == pkg_name for pkg in installed_pkgs):
-        shakedown.install_package(pkg_name, wait_for_completion=True)
+        options = {}
```
Comment (mgummelt) [inline tests/test.py]: This isn't guaranteed to be correct for an arbitrary package, so let's not include these options in this function.  Pass them as an argument to the function through `_require_package('hdfs')`.

Also, let's create the options functionally:
```
if (...):
  options = ...
else:
  options = ...
```

File: tests/test.py [hunk 2]
```diff
@@ -154,6 +158,11 @@ def _run_tests(app_url, app_args, expected_output, args={}, config={}):
 
 
 def _submit_job(app_url, app_args, args={}, config={}):
+    if os.environ.get('SECURITY') == 'strict':
```
Comment (mgummelt) [inline tests/test.py]: let's factor this out into an `_is_strict()` function, and replace all 3 occurrences

File: tests/test.py (hunk 1 above)
Comment (susanxhuynh) [inline tests/test.py]: Done

File: tests/test.py (hunk 2 above)
Comment (susanxhuynh) [inline tests/test.py]: Done

File: bin/test.sh [hunk 3]
```diff
@@ -64,7 +64,7 @@ setup_permissions() {
     if [ "$SECURITY" = "strict" ]; then
         # custom configuration to enable auth stuff:
         ${COMMONS_TOOLS_DIR}/setup_permissions.sh nobody "*" # spark's default service.role
-        ${COMMONS_TOOLS_DIR}/setup_permissions.sh nobody hdfs-role
+        ${COMMONS_TOOLS_DIR}/setup_permissions.sh root hdfs-role
```
Comment (mgummelt) [inline bin/test.sh]: @mohitsoni, @susanxhuynh says you instructed her to setup these permissions for root in hdfs in strict mode.  Why?  Aren't root users forbidden in strict mode?

**Pull request merged**